<launch>
    <arg name="threshold" default="0.425" />
    <arg name="viz_topic" default="/subjects/blink_images" />
    <arg name="viz" default="True" />
    <arg name="blink_backend" default="pytorch" /> <!-- can be tensorflow or pytorch -->
    <arg name="device_id_blinkestimation"  default="cuda:0" /> <!-- format depends on backend, e.g. /gpu:0 or /cpu:0 for tensorflow, or cuda:0 or cpu:0 for pytorch -->
    <arg name="model_type" default="vgg16" /> <!-- vgg16 is default for pytorch, and densnet121 is for tensorflow -->

    <node pkg="rt_gene" type="estimate_blink.py" name="estimate_blink" output="screen">
        <rosparam param="model_files">['model_nets/blink_model_pytorch_vgg16_allsubjects1.model','model_nets/blink_model_pytorch_vgg16_allsubjects2.model']</rosparam>
        <param name="blink_backend" value="$(arg blink_backend)" />
        <param name="threshold"  value="$(arg threshold)" />
        <param name="viz_topic"  value="$(arg viz_topic)" />
        <param name="viz"  value="$(arg viz)" />
        <param name="device_id_blinkestimation"  value="$(arg device_id_blinkestimation)" />
        <param name="model_type" value="$(arg model_type)" />
    </node>
</launch>
